cmd/cluster-olm-operator/main.go

```
@@ -366,6 +373,27 @@ func runOperator(ctx context.Context, cc *controllercmd.ControllerContext) error
 
 	operatorLoggingController := loglevel.NewClusterOperatorLoggingController(cl.OperatorClient, cc.EventRecorder.ForComponent("ClusterOLMOperatorLoggingController"))
 
+	// Watch for infrastructure topology changes. Topology changes are exceedingly rare
+	// (e.g., SNO to HA conversion) but require re-rendering the Helm manifests with the
+	// correct replica count and PDB settings. Exiting causes the deployment controller to
+	// restart cluster-olm-operator, which re-renders the manifests on startup.
+	initialTopology := infra.Status.ControlPlaneTopology
+	if _, err := cl.InfrastructureClient.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(old, new interface{}) {
```

golangci-lint (2.12.1)

[warning] redefines-builtin-id: redefinition of the built-in function new

(revive)